cpp/arcticdb/version/symbol_list.cpp

```diff
@@ -166,13 +177,24 @@ bool SymbolList::can_update_symbol_list(const std::shared_ptr<Store>& store,
             const folly::Range<KeyVectorItr>& keys) {
         SYMBOL_LIST_RUNTIME_LOG("Loading symbols from symbol list keys");
         bool read_compaction = false;
+        uint64_t uncompacted_keys_found = 0;
         CollectionType symbols{};
         for(const auto& key : keys) {
             if(key.id() == compaction_id) {
                 read_list_from_storage(store, key, symbols);
                 read_compaction = true;
             }
             else {
+                uncompacted_keys_found++;
+                if (uncompacted_keys_found == this->max_delta_ * 2 && !warned_expected_slowdown_) {
```

Comment:
This is wrong. This branch is processed entirely in RAM with no possibility of any noticeable slowdown.

Reply:
This has to go one level higher, in other words? Line 34?

Reply:
This is the "fast"/"cached" code path, so I don't think any warning is necessary in this code path. This is as fast as the user can possibly get for `list_symbols`.

Reply:
https://github.com/man-group/ArcticDB/pull/500